File: platform/frontend/src/api/streaming-preview.ts
import { authFetch } from './auth'

export interface StreamingPreviewSession {
  id: string
  devRequestId: string | null
  sessionName: string
  status: string
  streamType: string
  totalTokens: number
  streamDurationMs: number
  chunksDelivered: number
  createdAt: string
  updatedAt: string
}

export interface StreamingPreviewDetail {
  id: string
  devRequestId: string | null
  sessionName: string
  status: string
  streamType: string
  generatedCode: string | null
  previewHtml: string | null
  reasoningSteps: string[]
  totalTokens: number
  streamDurationMs: number
  chunksDelivered: number
  actions: string[]
  createdAt: string
  updatedAt: string
}

export interface StreamingPreviewStats {
  totalSessions: number
  byType: Record<string, number>
  avgDurationMs: number
  totalTokens: number
  streamsCompleted: number
}

export async function listStreamingPreviewSessions(): Promise<{ sessions: StreamingPreviewSession[] }> {
  const res = await authFetch('/api/streaming-preview/sessions')
  if (!res.ok) throw new Error('Failed to fetch streaming preview sessions')
  return res.json()
}

export async function getStreamingPreviewSession(id: string): Promise<StreamingPreviewDetail> {
  const res = await authFetch(`/api/streaming-preview/sessions/${id}`)
  if (!res.ok) throw new Error('Failed to fetch streaming preview session')
  return res.json()
}

export async function createStreamingPreviewSession(data: {
  name: string
  streamType: string
  devRequestId?: string
}): Promise<StreamingPreviewDetail> {
  const res = await authFetch('/api/streaming-preview/sessions', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  })
  if (!res.ok) throw new Error('Failed to create streaming preview session')
  return res.json()
}

export async function startStreamingPreview(id: string): Promise<StreamingPreviewDetail> {
  const res = await authFetch(`/api/streaming-preview/sessions/${id}/start`, {
    method: 'POST',
  })
  if (!res.ok) throw new Error('Failed to start streaming preview')
  return res.json()
}

export async function executeStreamingPreviewAction(
  id: string,
  action: string
): Promise<{ success: boolean; message: string }> {
  const res = await authFetch(`/api/streaming-preview/sessions/${id}/action`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ action }),
  })
  if (!res.ok) throw new Error('Failed to execute action')
  return res.json()
}

export async function getStreamingPreviewStats(): Promise<StreamingPreviewStats> {
  const res = await authFetch('/api/streaming-preview/stats')
  if (!res.ok) throw new Error('Failed to fetch streaming preview stats')
  return res.json()
}

File: platform/backend/AiDevRequest.API/Entities/StreamingPreview.cs
using System.ComponentModel.DataAnnotations;

namespace AiDevRequest.API.Entities;

public class StreamingPreview
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(100)]
    public required string UserId { get; set; }

    public Guid? DevRequestId { get; set; }

    [MaxLength(200)]
    public string SessionName { get; set; } = "";

    [MaxLength(20)]
    public string Status { get; set; } = "idle";

    [MaxLength(50)]
    public string StreamType { get; set; } = "code";

    public string? GeneratedCode { get; set; }

    public string? PreviewHtml { get; set; }

    public string? ReasoningStepsJson { get; set; }

    public int TotalTokens { get; set; } = 0;

    public int StreamDurationMs { get; set; } = 0;

    public int ChunksDelivered { get; set; } = 0;

    public string? ActionsJson { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
